packages/ui/apis/axiosInstance.ts

```diff
@@ -0,0 +1,61 @@
+import axios, { AxiosInstance, AxiosRequestConfig } from 'axios';
+
+import {
+  onErrorRequest,
+  onErrorResponse,
+  onRequest,
+  onResponse,
+} from './axiosInterceptor';
+
+const BASE_URL = '/example';
+
+class AxiosService {
+  private static instance: AxiosInstance;
+
+  private constructor() {}
+
+  private static getInstance(): AxiosInstance {
+    if (!AxiosService.instance) {
+      AxiosService.instance = axios.create({ baseURL: BASE_URL });
+      AxiosService.addInterceptors(AxiosService.instance);
+    }
+    return AxiosService.instance;
+  }
+
+  private static addInterceptors(instance: AxiosInstance) {
+    instance.interceptors.request.use(onRequest, onErrorRequest);
+    instance.interceptors.response.use(onResponse, onErrorResponse);
+  }
+
+  public static async get<T, U>(
+    url: string,
+    config: AxiosRequestConfig<U>,
+  ): Promise<T> {
+    const instance = AxiosService.getInstance();
+    const response = await instance.get<T>(url, config);
+    return response.data;
+  }
+
+  public static async post<T, U>(url: string, data: U): Promise<T> {
+    const instance = this.getInstance();
+    const response = await instance.post<T>(url, data);
+    return response.data;
```

Comment:
p5: Axios instance 의 `get`, `post`, `delete`, `patch` 는 `Promise`  객체를 반환하므로, 이렇게 반환해도 괜찮지 않을까요?
```suggestion
    return instance.post<T>(url, data);
```

Reply:
response에서 data만 뽑아내서 사용하고 싶어서 의도적으로 await를 사용했습니다!

Reply:
p4: 오 그렇군요! 제 생각에는 `response` 에서 `data` 를 뽑아내는 모든 메서드마다 존재하므로, response interceptor 에서 `data` 를 뽑아내는 로직을 추가하면 코드가 좀 더 명확해질 것 같습니다!

Reply:
response interceptor를 제가 생각하지 못했네요 ㅋㅋ 반영하겠습니다